src/lib/ui.ts

```diff
@@ -15,9 +15,22 @@ export function shorten(
   front: number = 3,
   back: number = 3
 ): string {
-  return `${text.slice(0, front)}…${text.slice(-back)}`
+  return `${text?.slice(0, front)}…${text?.slice(-back)}`
```

Comment:
`text` is not optional here so no need for the `?`
```suggestion
  return `${text.slice(0, front)}…${text.slice(-back)}`
```